lms/views/basic_lti_launch.py

```diff
@@ -227,3 +232,13 @@ def configure_module_item(self):
         self.context.js_config.maybe_enable_grading()
 
         return {}
+
+
+def record_course(context, request):
+    """Add the current course to the course table in the DB."""
+    Course.insert_if_not_exists(
+        request.db,
+        context.h_group.authority_provided_id,
+        request.lti_user.oauth_consumer_key,
+        {"canvas": {"sections_enabled": context.canvas_sections_enabled}},
+    )
```

Comment:
I think this might belong with the `insert_if_not_exists` in a course service. It would then be accessible from `content_item_selection` more naturally without being imported from here.

It would also keep the logic in one place. Currently it's split between here and the DB model.

I was surprised to not see the settings not being wholesale copied from the application instance here.

Reply:
Would we want to assume that _all_ application instance settings (now and going forward) are overridable on a per-course basis?

Reply:
Which ever way you go will be wrong in a different circumstance, so we can't know which is the right path. Both could change but it makes the rule simple for now? It's easier to think about.